Neos.Neos/Tests/Behavior/Features/Bootstrap/FeatureContext.php

```diff
@@ -102,4 +109,27 @@ protected function createContentRepository(
 
         return $contentRepository;
     }
+
+    protected function deserializeProperties(array $properties): PropertyValuesToWrite
+    {
+        $properties = array_map(
+            $this->loadFlowObjectsRecursive(...),
+            $properties
+        );
+
+        return $this->deserializePropertiesCrTestSuiteTrait($properties);
+    }
+
+    public function loadFlowObjectsRecursive(mixed $value): mixed
+    {
+        if (is_array($value) && isset($value['__flow_object_type'])) {
+            return $this->persistenceManager->getObjectByIdentifier($value['__identifier'], $value['__flow_object_type'], true);
+        } elseif (is_array($value)) {
+            return array_map(
+                $this->loadFlowObjectsRecursive(...),
+                $value
+            );
+        }
+        return $value;
+    }
```

Comment:
nice trick to fully allow the flow object syntax `{"__flow_object_type":"Neos\\Media\\Domain\\Model\\Asset","__identifier":"an-asset-to-change"}`

i would slightly prefer the use of a custom synax to not confuse it with the final serialization format like

```json
{
"uriPathSegment": "a1-1",
"title": "Node a1-1",
"asset": "Asset:an-asset-to-change"
}
```

like we do in https://github.com/neos/neos-development-collection/blob/d0f4eccbb0c7c9b8d224008001a5fb299b516d50/Neos.ContentRepository.BehavioralTests/Tests/Behavior/Features/02-NodeCreation/05-CreateNodeAggregateWithNode_ComplexDefaultAndInitialProperties.feature#L98-L115 but this is certainly more versatile and cool :D

Reply:
I was thinking about that.... But why creating a new syntax if we already have one? And how would you handle arrays of assets?

```
{
"uriPathSegment": "a1-1",
"title": "Node a1-1",
"assets": "Assets:an-asset-to-change,some-other-asset"
}
```

Reply:
yeah exactly that are the things that speak against this ;) only by using a know serialisation format one might think by reading the tests that we write directly to the db which is the opposite.

Reply:
> But why creating a new syntax if we already have one?

The one we have is an implementation detail of the Flow persistence really.
Since Behat tests are also meant to be _read_ by humans, I agree to @mhsdesign 

```gherkin
And the following CreateNodeAggregateWithNode commands are executed:
      | nodeAggregateId | parentNodeAggregateId | nodeTypeName                 | initialPropertyValues                                                                                                                                        | nodeName |
      | a               | root                  | Neos.Neos:Site               | {}                                                                                                                                                           | site     |
      | a1              | a                     | Neos.Neos:Test.DocumentType1 | {"uriPathSegment": "a1", "title": "Node a1", "asset": {"__flow_object_type":"Neos\\Media\\Domain\\Model\\Asset","__identifier":"an-asset-to-change"}}        | a1       |
      | a1-1            | a1                    | Neos.Neos:Test.DocumentType1 | {"uriPathSegment": "a1-1", "title": "Node a1-1", "assets": [{"__flow_object_type":"Neos\\Media\\Domain\\Model\\Asset","__identifier":"an-asset-to-change"}]} | a1-1     |
      | a1-2            | a1                    | Neos.Neos:Test.DocumentType1 | {"uriPathSegment": "a1-2", "title": "Node a1-2", "asset": {"__flow_object_type":"Neos\\Media\\Domain\\Model\\Asset","__identifier":"some-other-asset"}}      | a1-2     |
      | a2              | a                     | Neos.Neos:Test.DocumentType2 | {"uriPathSegment": "a2", "title": "Node a2", "text": "Link to asset://an-asset-to-change."}                                                                  | a2       |
      | a3              | a                     | Neos.Neos:Test.DocumentType2 | {"uriPathSegment": "a2", "title": "Node a2", "text": "Link to asset://some-other-asset."}                                                                    | a3       |
```

is much harder to read than

```gherkin
And the following CreateNodeAggregateWithNode commands are executed:
      | nodeAggregateId | parentNodeAggregateId | nodeTypeName                 | initialPropertyValues                                                                                                                                        | nodeName |
      | a               | root                  | Neos.Neos:Site               | {}                                                                                                                                                           | site     |
      | a1              | a                     | Neos.Neos:Test.DocumentType1 | {"uriPathSegment": "a1", "title": "Node a1", "asset": "Asset:an-asset-to-change"}        | a1       |
      | a1-1            | a1                    | Neos.Neos:Test.DocumentType1 | {"uriPathSegment": "a1-1", "title": "Node a1-1", "assets": ["Asset:an-asset-to-change"]} | a1-1     |
      | a1-2            | a1                    | Neos.Neos:Test.DocumentType1 | {"uriPathSegment": "a1-2", "title": "Node a1-2", "asset": "Asset:some-other-asset"}      | a1-2     |
      | a2              | a                     | Neos.Neos:Test.DocumentType2 | {"uriPathSegment": "a2", "title": "Node a2", "text": "Link to asset://an-asset-to-change."}                                                                  | a2       |
      | a3              | a                     | Neos.Neos:Test.DocumentType2 | {"uriPathSegment": "a2", "title": "Node a2", "text": "Link to asset://some-other-asset."}                                                                    | a3       |
```

But obviously this could be done in a follow-up

Reply:
Another point that speaks _for_ this is that when its easier to find where the custom `Asset:an-asset-to-change` logic is implemented. As theres likely a regex or string reference to `/Asset:[0-9a-z-]+/` in the behat code base.

The way i see it the `ClassName:identifier` is our special syntax that means _insert a real php object instance in here, but i can only use json in behat_ ^^

so +1 for a followup ;)

Reply:
https://github.com/neos/neos-development-collection/pull/5029